docs/DeveloperGuide.md

```diff
@@ -626,7 +626,161 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+**Examples**
+
+- find `John` returns contact with `name` of `John Doe` and corresponding `IC_NUMBER` of `T0123456A`
+- find `johndoe@mail.com` returns contact with `email` of `johndoe@mail.com` and corresponding `IC_NUMBER` of `T0123456A`
+
+### Update notes window during execution of `find` command
+
+**Potential Issues With Current Feature**
+
+Users have given us the feedback that our notes window on the right hand side of the split pane does not update with the use of `find` command.
+For instance, if our notes panel currently already have `John`'s notes reflected on it, using the `find` command to find someone who is not `John` will not cause the 
+notes panel to update accordingly. `John`'s notes will still exist on the notes panel despite the executing the `find` command to `find` a differing person.
+This could bring inconvenience to the user as users will not be able to view a contact's notes immediately upon using the `find` command. 
+They would then have to go through the hassle of typing the `show` command to allow the new contact's note to be displayed on the notes panel.
+
+**Proposed Enhancements**
+
+In order to make it more convenient for the users, we will enhance `find` command such that on executing the `find` command, the notes of the first 
+person appearing at the top of the search results will replace the current notes in the notes panel. We will also allow users to know whose note they are viewing, by displaying the patient's name or IC number.
+
+**Examples**
+
+- If the notes panel is currently displaying the notes of the user with `IC_NUMBER` of `T1234567C`, 
+executing `find T0123456A`, will cause the notes panel to update to the notes of user with `IC_NUMBER` of `T0123456A`
+
+### Enabling optional fields
+
+**Potential Issues With Current Feature**
+
+Currently, all fields: `name`, `age`, `sex`, `ic`, `phone`, `email` and `address` are compulsory. However, there might be instances in which
+`phone` and `email` might not be necessary since patients might not provide. 
+
+**Proposed Enhancements**
+
+We will be allowing some fields such as `phone` and `email` which are not necessary to be optional. 
+
+**Examples**
+
+- `add n\John Doe p\12345678 i\T0123456A ag\12 s\M a\311, Clementi Ave 2, #02-25` will be allowed.
+- `add n\John Doe e\johndoe@mail.com i\T0123456A ag\12 s\M a\311, Clementi Ave 2, #02-25` will be allowed.
+- `add n\John Doe i\T0123456A ag\12 s\M a\311, Clementi Ave 2, #02-25` will be allowed.
+
+### Improving error handling for `edit` command
+
+**Potential Issues With Current Feature**
+
+Currently, the `edit` command does not check the existence of `IC_NUMBER` upon entering an `edit` command with an empty field.
+For instance, entering `edit S1234567P` in the event that the contact with the `IC_NUMBER` of `S1234567P` do not exist returns an error message 
+of *at least one field to edit must be provided*. However, the error message should be such that *the IC_NUMBER provided does
```

Comment:
can remove the 'such'

Reply:
Capitalise the at for 'At least one....' and the for 'The `IC_NUMBER` provided...' can

The IC_NUMBER also put in code block!